syncapi/streams/stream_pdu.go

```diff
@@ -519,7 +519,7 @@ func (p *PDUStreamProvider) updatePowerLevelEvent(ctx context.Context, ev *rstyp
 	}
 
 	var evNew gomatrixserverlib.PDU
-	evNew, err = gomatrixserverlib.MustGetRoomVersion(gomatrixserverlib.RoomVersionPseudoIDs).NewEventFromTrustedJSON(newEv, false)
+	evNew, err = gomatrixserverlib.MustGetRoomVersion(gomatrixserverlib.RoomVersionPseudoIDs).NewEventFromTrustedJSONWithEventID(ev.EventID(), newEv, false)
```

Comment:
Oh, oops, good find!

Reply:
This one took a long time to diagnose! It would send sytest into loops whilst it waits for a previously sent event ID in the sync response - but only for power level events on pseudo ID rooms. Eventually nailed it down to this though, it's always the one-line fixes that come after hours of debugging